### PR TITLE
Luc

### PR DESCRIPTION
**Pull request by lesevereijns**

### CODE REVIEW

**Comment by lesevereijns** [comment]
multiwords & term elements

File: tests/test_nafdocument.py [hunk 1]
```diff
@@ -5,7 +5,8 @@
 from collections import namedtuple
 
 from nafigator.nafdocument import NafDocument
-from nafigator import nafdocument, EntityElement
+from nafigator import nafdocument, EntityElement, TermElement, MultiwordElement
+from collections import namedtuple
```
Comment (chanfk) [inline tests/test_nafdocument.py:9]: This line is a duplicate of line 5

Comment (lesevereijns) [reply tests/test_nafdocument.py]: check

File: tests/test_nafdocument.py [hunk 2]
```diff
@@ -254,23 +248,20 @@ def test_add_public_element(self, doc: NafDocument, public_var: str):
...
-        #     element=doc.layer("text"),
-        #     tag="wf",
-        #     data=WF_ELEMENT,
-        #     attributes_to_ignore=["text"],
-        # ) 
-        
-        # # fails on dict and on element input
-        # doc.add_wf_element(wf,True)
-        pass
+        doc = NafDocument().open(r"tests/tests/example.naf.xml")
```
[the remark this answers is not in the document]
Comment (lesevereijns) [reply tests/test_nafdocument.py]: check

File: tests/test_nafdocument.py [hunk 3]
```diff
@@ -39,16 +40,9 @@ def public_var():
...
-        "text" : "test_text",
-        "id" : "test_id",
-        "sent" : "test_sent",
-        "para" : "test_para",
-        "page" : "test_page",
-        "offset" : "test_offset",
-        "length" : "test_length",
-        "xpath" : "test_xpath",
-    }
+    wf_element = namedtuple("WfElemenwt","text id sent para page offset length xpath")
```
[the remark this answers is not in the document]
Comment (lesevereijns) [reply tests/test_nafdocument.py:42]: true

File: tests/test_nafdocument.py [hunk 4]
```diff
@@ -254,23 +248,20 @@ def test_add_public_element(self, doc: NafDocument, public_var: str):
...
-        #     tag="wf",
-        #     data=WF_ELEMENT,
-        #     attributes_to_ignore=["text"],
-        # ) 
-        
-        # # fails on dict and on element input
-        # doc.add_wf_element(wf,True)
-        pass
+        doc = NafDocument().open(r"tests/tests/example.naf.xml")
+        doc.add_wf_element(wf_element_var,True)
```
Comment (hmduerinck) [inline tests/test_nafdocument.py:257]: afdekken cdata=False
cdata=true comineren met ']]>' in text en zonder

Comment (lesevereijns) [reply tests/test_nafdocument.py]: done

File: tests/test_nafdocument.py [hunk 5]
```diff
@@ -359,14 +395,43 @@ def test_add_external_reference_element(self):
...
+        doc.add_multiword_element(data)
+
+        find_entities = doc.find(nafdocument.MULTIWORDS_LAYER_TAG).find(f"./{nafdocument.MULTIWORD_OCCURRENCE_TAG}[@id='test_id']")
+        assert find_entities.attrib == {"id": test_id, "type": test_type}
+        
+        if span != []:
+            doc.add_span_element.assert_called_once()
+
+        
+
```
Comment (royklip) [inline tests/test_nafdocument.py:434]: Whitespace

File: tests/test_nafdocument.py [hunk 6]
```diff
@@ -359,14 +395,43 @@ def test_add_external_reference_element(self):
...
+            case=None,
+            status = None,
+            components= components,
+        )
+
+
+        doc.add_span_element = MagicMock()
+        doc.add_multiword_element(data)
+
+        find_entities = doc.find(nafdocument.MULTIWORDS_LAYER_TAG).find(f"./{nafdocument.MULTIWORD_OCCURRENCE_TAG}[@id='test_id']")
```
Comment (royklip) [inline tests/test_nafdocument.py:427]: [@id='test_id'] can be replaced by [@id='{test_id}']

File: tests/test_nafdocument.py [hunk 7]
```diff
@@ -323,14 +326,47 @@ def test_add_entity_element(self, doc: NafDocument, span: list, ext_refs: list):
...
+            span = span,
+            ext_refs=ext_refs,
+            comment=None
+        )
+
+        doc.add_span_element = MagicMock()
+        doc.add_external_reference_element = MagicMock()
+        doc.add_term_element(data, layer_to_attributes_to_ignore={},comments=False)
+
+        find_terms = doc.find(nafdocument.TERMS_LAYER_TAG).find(f"./{nafdocument.TERM_OCCURRENCE_TAG}[@id='test_id']")
```
Comment (royklip) [inline tests/test_nafdocument.py:362]: [@id='test_id'] can be replaced by [@id='{test_id}']

**Pull request merged**